projects/plugins/jetpack/extensions/plugins/sharing/sharing.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Block Editor - Sharing feature.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Sharing;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Status;
+use Automattic\Jetpack\Status\Host;
+use Jetpack_Gutenberg;
+
+/**
+ * Register Sharing plugin.
+ *
+ * @return void
+ */
+function register_plugins() {
+	// Connection check.
+	if (
+		( new Host() )->is_wpcom_simple()
+		|| ( ( new Connection_Manager( 'jetpack' ) )->has_connected_owner() && ! ( new Status() )->is_offline_mode() )
+	) {
```

Comment:
I don't know if we need this here; the sharing buttons can be used without a connection to WordPress.com, in offline mode.

Reply:
Thanks!